tests/SourcemapToolkit.SourcemapParser.UnitTests/SourceMapUnitTests.cs

```diff
@@ -118,5 +119,230 @@ public void GetMappingEntryForGeneratedSourcePosition_NoExactMatchHasSimilarOnDi
             // Asset
             Assert.AreEqual(matchingMappingEntry, result);
         }
-    }
+
+		private MappingEntry getSimpleEntry(SourcePosition generatedSourcePosition, SourcePosition originalSourcePosition, string originalFileName)
+		{
+			return new MappingEntry
+			{
+				GeneratedSourcePosition = generatedSourcePosition,
+				OriginalSourcePosition = originalSourcePosition,
+				OriginalFileName = originalFileName
+			};
+		}
+
+		private SourcePosition generateSourcePosition(int lineNumber, int colNumber = 0)
+		{
+			return new SourcePosition
+			{
+				ZeroBasedLineNumber = lineNumber,
+				ZeroBasedColumnNumber = colNumber
+			};
+		}
+
+		[TestMethod]
+		public void GetRootMappingEntryForGeneratedSourcePosition_NoChildren_ReturnsSameEntry()
+		{
+			// Arrange
+			SourcePosition generated1 = generateSourcePosition(2);
+			SourcePosition original1 = generateSourcePosition(1);
+			MappingEntry mappingEntry = getSimpleEntry(generated1, original1, "generated.js");
+
+			SourceMap sourceMap = new SourceMap
+			{
+				Sources = new List<string> { "generated.js" },
+				ParsedMappings = new List<MappingEntry> { mappingEntry }
+			};
+
+			// Act
+			MappingEntry rootEntry = sourceMap.GetMappingEntryForGeneratedSourcePosition(generated1);
+
+			// Assert
+			Assert.AreEqual(rootEntry, mappingEntry);
+		}
+
+		[TestMethod]
+		[ExpectedException(typeof(ArgumentNullException))]
+		public void ApplyMap_NullSubmap_ThrowsException()
+		{
+			// Arrange
+			SourcePosition generated2 = generateSourcePosition(3);
```

Comment:
put an inline comment here saying what the paremeter means, something like (3 /* lineNumber */)